easybuild/toolchains/compiler/craypewrappers.py

```diff
@@ -0,0 +1,182 @@
+    }
+
+    COMPILER_CC = 'cc'
+    COMPILER_CXX = 'CC'
+
+    COMPILER_F77 = 'ftn'
+    COMPILER_F90 = 'ftn'
+
+    # FIXME (kehoste) hmmmm, really? then how do you control optimisation, precision when using the Cray wrappers?
+    COMPILER_FLAGS = []  # we dont have this for the wrappers
+    COMPILER_OPT_FLAGS = []  # or those
+    COMPILER_PREC_FLAGS = []  # and those for sure not !
+
+    # template and name suffix for PrgEnv module that matches this toolchain
+    # e.g. 'gnu' => 'PrgEnv-gnu/<version>'
+    PRGENV_MODULE_NAME_TEMPLATE = 'PrgEnv-%(suffix)s/%(version)s'
+    PRGENV_MODULE_NAME_SUFFIX = None
+
+    # template for craype module (determines code generator backend of Cray compiler wrappers)
+    CRAYPE_MODULE_NAME_TEMPLATE = 'craype-%(optarch)s'
+
+    def _pre_preprare(self):
+        """Load PrgEnv module."""
+        prgenv_mod_name = self.PRGENV_MODULE_NAME_TEMPLATE % {
+            'suffix': self.PRGENV_MODULE_NAME_SUFFIX,
+            'version': self.version,
+        }
+        self.log.info("Loading PrgEnv module '%s' for Cray toolchain %s" % (prgenv_mod_name, self.mod_short_name))
+        self.modules_tool.load([prgenv_mod_name])
+
+    def _set_optimal_architecture(self):
+        """Load craype module specified via 'optarch' build option."""
+        optarch = build_option('optarch')
+        if optarch is None:
+            self.log.error("Don't know which 'craype' module to load, 'optarch' build option is unspecified.")
+        else:
+            self.modules_tool.load([self.CRAYPE_MODULE_NAME_TEMPLATE % {'optarch': optarch}])
+
+    # FIXME: (kehoste) is it really needed to customise this?
+    # this looks like a workaround for setting the COMPILER_*_FLAGS lists empty?
+    def _set_compiler_flags(self):
+        """Collect the flags set, and add them as variables too"""
+
+        flags = [self.options.option(x) for x in self.COMPILER_FLAGS if self.options.get(x, False)]
+        cflags = [self.options.option(x) for x in self.COMPILER_C_FLAGS + self.COMPILER_C_UNIQUE_FLAGS \
+                  if self.options.get(x, False)]
+        fflags = [self.options.option(x) for x in self.COMPILER_F_FLAGS + self.COMPILER_F_UNIQUE_FLAGS \
+                  if self.options.get(x, False)]
+
+
+        # precflags last
+        self.variables.nappend('CFLAGS', flags)
+        self.variables.nappend('CFLAGS', cflags)
+
+        self.variables.nappend('CXXFLAGS', flags)
+        self.variables.nappend('CXXFLAGS', cflags)
+
+        self.variables.nappend('FFLAGS', flags)
+        self.variables.nappend('FFLAGS', fflags)
+
+        self.variables.nappend('F90FLAGS', flags)
+        self.variables.nappend('F90FLAGS', fflags)
+
+
+# Gcc's base is Compiler
+class CrayPEWrapperGNU(CrayPEWrapper):
+    """Support for using the Cray GNU compiler wrappers."""
+    TC_CONSTANT_CRAYPEWRAPPER = TC_CONSTANT_CRAYPEWRAPPER + '_GNU'
+
+    PRGENV_MODULE_NAME_SUFFIX = 'gnu'  # PrgEnv-gnu
+
+    def _set_compiler_vars(self):
```

Comment:
make a new private method in `CrayPEWrapper` that does this so you can share it with Intel and call it here like
```python
def _set_compiler_vars(self):
    self._new_method(Gcc, [])
    super(CrayPEWrapperGNU,self)._set_compiler_vars()
```
and for intel
def _set_compiler_vars(self):
    self._new_method(IntelIccIfort, ['LINKER_TOGGLE_STATIC_DYNAMIC'])
    super(CrayPEWrapperGNU,self)._set_compiler_vars()
```
(generating and reviewing almost exact copy/paste code is hard :smile:)